core/lib/dal/src/connection/mod.rs

```diff
@@ -307,6 +307,12 @@ impl ConnectionPool {
         self.max_size
     }
 
+    /// Uses this pool to report Postgres-wide metrics (e.g., table sizes). Should be called sparingly to not spam
+    /// identical metrics from multiple places. The returned future runs indefinitely and should be spawned as a Tokio task.
+    pub async fn run_postgres_metrics_scraping(self, scrape_interval: Duration) {
```

Comment:
I have first tried using a [`Collector`](https://matter-labs.github.io/vise/vise/struct.Collector.html) to scrape these metrics, but since table sizes are not changed *that* rapidly, I think that updating the corresponding metrics on a schedule would be preferable.